README.rst

```diff
@@ -454,16 +454,184 @@ template ``context_processors`` in ``settings.py`` as shown below.
+    }
+    register_menu_group(position=1, config=my_model_list_config)
+    register_menu_group(position=2, config=my_model_add_config)
+
+Following is the description of the configuration:
+
++------------------+--------------------------------------------------------------+
+| **Parameter**    | **Description**                                              |
++------------------+--------------------------------------------------------------+
+| ``model``        | (``str``) Model of a app.  (``required``)                    |
++------------------+--------------------------------------------------------------+
+| ``name``         | (``str``) url name. eg. changelist or add.  (``required``)   |
++------------------+--------------------------------------------------------------+
+| ``label``        | (``str``) Display name for the link.  (``optional``)         |
++------------------+--------------------------------------------------------------+
+| ``icon``         | (``str``) CSS class name of the link icon.  (``optional``)   |
++------------------+--------------------------------------------------------------+
+
+Create a menu group
+^^^^^^^^^^^^^^^^^^^
+
+If you want to create a nested menu group then you can use following syntax.
+It creates a dropdown in the menu.
+
+**Syntax:**
+
+.. code-block:: python
+    
+    my_group_items = {
+        1: {'label': 'Link Label', 'url': 'link_url', 'icon': 'my-icon'},
+        2: {
+            'model': 'my_project.MyModel',
+            'name': 'changelist',
+            'label': 'MyModel Change List',
+            'icon': 'my-model-list-class',
+            }
+    }
+    register_menu_group(
+        position=1,
+        config={'label': 'My Label', 'items': my_group_items, 'icon': 'my-icon-class'},
+    )
+
+Following is the description of the configuration:
+
++------------------+--------------------------------------------------------------+
+| **Parameter**    | **Description**                                              |
++------------------+--------------------------------------------------------------+
+| ``label``        | (``str``) Display name for the link.  (``required``)         |
++------------------+--------------------------------------------------------------+
+| ``items``        | (``dict``) Items to be displayed in the dropdown.            |
+|                  | It can be a dict of ``general link`` or ``model link``       |
+|                  | with key as their position.                                  |
+|                  | (``required``)                                               |
++------------------+--------------------------------------------------------------+
+| ``icon``         | (``str``) CSS class name of the group icon.  (``optional``)  |
++------------------+--------------------------------------------------------------+
+
+**Syntax to add icon using class name:**
+
+.. code-block:: css
+
+    .icon-class-name:{
+        background: url(imageurl);
+    }
```

Comment:
I don't understand where this CSS should be added, I guess you're implying it can be done using with custom additional CSS files, can you make this more explicit please? And link the specific README section regarding custom CSS if needed.

Reply:
We are already having a section for it. https://github.com/openwisp/openwisp-utils#supplying-custom-css-and-js-for-the-admin-theme. Should create one more add a subsection in this?
I think we can create a subsection in Main Navigation Menu section

Reply:
I would turn this in a subsection with the title "How to use custom icons in the menu", then explain that it's just needed to supply a custom CSS as indicated in "Supplying custom CSS and JS for the admin theme" (make this a link) which defines the CSS classes with the custom icons.